### PR TITLE
Diagram based database

### PR DESCRIPTION
**Pull request by Vgonma**
# Changes
- Created schema_based_on_diagram.sql to create a database following the following diagram.
- Add an extra relation table **histories_treatments** for the many-to-many relationships between medical_histories and treatments.

clinic_diagram

### CODE REVIEW

**Review by BenMukebo — COMMENTED** [review]
## STATUS: INVALID ❌ 

Hi  @Vgonma,

I can't miss felicitating your efforts and energy, you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request 

- [x] You have submitted a project, that doesn't have commits from all members in your pair programming group. Kindly note that this is a pair programming activity and all the members should commit to the project.
- Note the next reviewer will check the commit history to make sure that everybody has contributed code on this Pull Request.


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me (@BenMukebo) in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **Does not count** into the code reviews limit.

**Comment by Vgonma** [comment]
https://github.com/Vgonma/vet-clinic/pull/7#pullrequestreview-1318134304 Fulfilled all members committed to this project.

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+);
+
+CREATE TABLE IF NOT EXISTS invoice_items (
+  id SERIAL PRIMARY KEY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT
+);
```
Comment (omarsalem7) [inline schema_based_on_diagram.sql]: - [x] All foreign keys in all tables suppose to have indexes created on them.
> For example, in the medical_histories table create index like this:
```SQL
CREATE INDEX ON medical_histories (patient_id);
```

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @Vgonma, @shahadat3669 😊, 

Good job so far! 👏
You have done well reaching this milestone point, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- All tables created based on schema diagram ✔️
- Correct git-flow ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql (hunk 1 above)
Comment (Vgonma) [reply schema_based_on_diagram.sql]: https://github.com/Vgonma/vet-clinic/pull/7#discussion_r1120666557 Added indexes for all foreign keys

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+CREATE INDEX idx_treatment_id
+ON histories_treatments(treatment_id ASC);
+
+CREATE TABLE IF NOT EXISTS invoice_items (
+  id SERIAL PRIMARY KEY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT,
+  treatment_id INT
```
Comment (omarsalem7) [inline schema_based_on_diagram.sql:51]: - [x] Great work for adding indexes for foreign keys but you forget to add indexes for invoice_id and treatment_id in the invoice_items table, kindly create indexes on them

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @Vgonma, @shahadat3669 😊, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

Almost near

 ## Highlights
✔️ All tables created based on the schema diagram
✔️ Correct git-flow

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by youmari — APPROVED** [review]
# Project Approved :trophy: :tada: 🟢 

Hi @Vgonma 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linters error.
:heavy_check_mark: Details PR title and description.
:heavy_check_mark: Good commit messages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U02FTTCCU2V)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**